### PR TITLE
Remove resultSet

### PR DESCRIPTION
**Pull request by ohadbitt**
**Breaking Changes:**
- Remove ResultSet interface

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ protected KustoResultSetTable(JSONObject jsonTable) throws KustoServiceQueryErro
         return currentRow;
     }
 
-    @Override
+
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:142]: I think the file needs re-formatting
Sometimes the `@Override` is replaced with a blank line, other times it doesn't
Make sure the space between methods stays consistent

**Comment by AsafMah** [comment]
Since this is a proper breaking change we will need to bump the major, right?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 2]
```diff
@@ -506,42 +415,26 @@ public BigDecimal getBigDecimal(int columnIndex) {
...
     }
 
-    @Override
     public void beforeFirst() {
         rowIterator = rows.iterator();
     }
 
-    @Override
-    public void afterLast() {
-        while (next()) ;
```
Comment (yogilad) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:541]: Why was this removed (not deprecated it not implemented)?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 3]
```diff
@@ -1018,314 +577,11 @@ public URL getURL(String columnName) throws SQLException {
...
-    @Override
-    public boolean isWrapperFor(Class<?> aClass) {
-        return false;
-    }
-
     public int count() {
         return rows.size();
     }
+
+    public boolean isNullAt(int columnIndex){
```
Comment (yogilad) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:584]: IsColumnNull or simply IsNull

**Review by yogilad — APPROVED** [review]
Some small comments

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 2 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:541]: I dont think its relevant to this SDK usage - i think the only reason to use this is if you provide two ways iterator - so you can go afterLast and do while(previous)

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 3 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:584]: took this from Spark - i think its more clear

Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java]: will take isNull

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 4]
```diff
@@ -255,27 +235,14 @@ public Double getDoubleObject(int columnIndex) {
...
-        if (get(columnIndex) == null) {
-            return null;
-        }
-
-        return (BigDecimal) get(columnIndex);
-    }
-
-    @Override
-    public byte[] getBytes(int columnIndex) {
+   public byte[] getBytes(int columnIndex) {
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:238]: nit: missing a space in indentation

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 5]
```diff
@@ -928,7 +479,7 @@ public LocalDateTime getKustoDateTime(String columnName) {
     /**
      * This will cut the date up to yyyy-MM-dd'T'HH:mm:ss.SSS
      */
-    @Override
+
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:482]: nit: line

**Review by AsafMah — APPROVED** [review]
